translator/internaldata/oc_to_resource.go

```diff
@@ -69,7 +87,7 @@ func ocNodeResourceToInternal(ocNode *occommon.Node, ocResource *ocresource.Reso
 	if ocNode != nil {
 		// Copy all Attributes.
 		for k, v := range ocNode.Attributes {
-			attrs.InsertString(k, v)
+			tracetranslator.UpsertStringToAttributeMap(k, v, attrs, true)
```

Comment:
Do we expect duplicates here that require an Upsert instead of Insert?

Reply:
Other places using this method are all upserts. Is there a reason we should create another method to support this one spot?

Reply:
Yes, let me clarify why I think so.

UpsertStringToAttributeMap does the smart decoding and tries to understand what the value is and if it looks like a JSON map tries to convert it to OTLP map. I don't think we should be doing it for OpenCensus format or any other format.

The only exception is Zipkin which already does this and I don't want to change that behavior because it appears this specific Zipkin behavior was contributed as bug fixes by other people and I don't want to break it.

We _should_ do only the translation in opposite direction, i.e. if we have an OTLP Array or Map and we need to convert to a format that does not allow representing an array or map then we need to flatten it as a string. We already do this and I think that is the reasonable thing to do, so we need to keep it.

BTW, I don't think we need another method. `InsertString` is sufficient.

Reply:
Reverted to original behavior